cmd/integration/harness.go

```diff
@@ -0,0 +1,194 @@
+		fmt.Println("Full node ERROR listening on:", server.cfg.rpcListen+"\n")
+		return nil, err
+	}
+
+	//// Verify the client connectivity.
+	//// If failed, shutdown the server.
+	//conn, err := connectClient(cfg.rpcListen)
+	//if err != nil {
+	//	_ = server.shutdown()
+	//	return nil, err
+	//}
+
+	h := &Harness{
+		server: server,
+		//conn:                   conn,
+		//SpacemeshServiceClient: pb.NewSpacemeshServiceClient(conn),
+	}
+
+	return h, nil
+}
+
+// TearDown stops the harness running instance.
+// The created process is killed
+func (h *Harness) TearDown() error {
+	if err := h.server.shutdown(); err != nil {
+		return err
+	}
+
+	if err := h.conn.Close(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// ProcessErrors returns a channel used for reporting any fatal process errors.
+func (h *Harness) ProcessErrors() <-chan error {
+	return h.server.errChan
+}
+
+// connectClient attempts to establish a gRPC Client connection
+// to the provided target.
+func connectClient(target string) (*grpc.ClientConn, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	opts := []grpc.DialOption{
+		grpc.WithInsecure(),
+		grpc.WithBlock(),
+	}
+	defer cancel()
+
+	conn, err := grpc.DialContext(ctx, target, opts...)
+	if err != nil {
+		return nil, fmt.Errorf("unable to connect to RPC server at %s: %v", target, err)
+	}
+	return conn, nil
+}
+
+// baseDir is the directory path of the temp directory for all the harness files.
+// NOTICE: right now it's not in use, might be useful in the near future
+func baseDir() (string, error) {
+	baseDir := filepath.Join(os.TempDir(), "full_node")
+	err := os.MkdirAll(baseDir, 0755)
+	return baseDir, err
+}
+
+func isListening(addr string) bool {
+	conn, _ := net.DialTimeout("tcp", addr, 1*time.Second)
+	if conn != nil {
+		_ = conn.Close()
+		return true
+	}
+	return false
+}
+
+func killProcess(address string) error {
+	addr, err := net.ResolveTCPAddr("tcp", address)
+	if err != nil {
+		return err
+	}
+
+	var cmd *exec.Cmd
+	if runtime.GOOS == "windows" {
+		args := fmt.Sprintf("(Get-NetTCPConnection -LocalPort %d).OwningProcess -Force", addr.Port)
+		cmd = exec.Command("Stop-Process", "-Id", args)
+	} else {
+		args := fmt.Sprintf("lsof -i tcp:%d | grep LISTEN | awk '{print $2}' | xargs kill -9", addr.Port)
+		cmd = exec.Command("bash", "-c", args)
+	}
+
+	var errb bytes.Buffer
+	cmd.Stderr = &errb
+
+	if err := cmd.Start(); err != nil {
+		return err
+	}
+
+	if err := cmd.Wait(); err != nil {
+		return fmt.Errorf("error during killing process: %s | %s", err, errb.String())
+	}
+
+	return nil
+}
+
+func main() {
+	// os.Args[0] contains the current process path
+	_, err := NewHarnessDefaultServerConfig(os.Args[1:])
+	if err != nil {
+		fmt.Println("An error has occurred while generating a new harness:", err)
+	}
+	// Sleeping here to enable the test run before harness terminates
+	fmt.Println("sleeping for 3000 seconds (50 mins)")
```

Comment:
why terminate? or why not terminate on signal?

Reply:
fixed in cb1cd48ca644f522e9d29fefd6f714012db20427